src/Plugin/GraphQL/Derivers/SearchAPIDocumentTypeDeriver.php

```diff
@@ -3,17 +3,32 @@
 namespace Drupal\graphql_search_api\Plugin\GraphQL\Derivers;
 
 use Drupal\Component\Plugin\Derivative\DeriverBase;
-use Drupal\search_api\Entity\Index;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\Discovery\ContainerDeriverInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Provides derivers for different index documents in Search API.
  */
-class SearchAPIDocumentTypeDeriver extends DeriverBase {
+class SearchAPIDocumentTypeDeriver extends DeriverBase implements ContainerDeriverInterface {
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
 
   /**
    * Constructs a new Search API document type.
    */
-  public function __construct() {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
```

Comment:
Here we should now replace the comment with the proper PHPDoc signature for the arguments in the constructor.
Something like : 
```
/**
 * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
 *   The entity type manager service.
 */
```